src/crypto/crypto_keys.cc

```diff
@@ -293,6 +292,25 @@ MaybeLocal<Value> WritePrivateKey(
   BIOPointer bio(BIO_new(BIO_s_mem()));
   CHECK(bio);
 
+  // If an empty string was passed as the passphrase, the ByteSource might
+  // contain a null pointer, which OpenSSL will ignore, causing it to invoke its
+  // default passphrase callback, which would block the thread until the user
+  // manually enters a passphrase. We could supply our own passphrase callback
+  // to handle this special case, but it is easier to avoid passing a null
+  // pointer to OpenSSL.
+  char* pass = nullptr;
+  size_t pass_len = 0;
+  if (!config.passphrase_.IsEmpty()) {
+    pass = const_cast<char*>(config.passphrase_->get());
+    pass_len = config.passphrase_->size();
+    if (pass == nullptr) {
+      // OpenSSL will not actually dereference this pointer, so it can be any
+      // non-null pointer.
+      CHECK_EQ(pass_len, 0);
+      pass = reinterpret_cast<char*>(-1);
```

Comment:
```suggestion
      static char dummy = 0;
      pass = &dummy;
```

(since `(char*)-1 != nullptr` is not technically guaranteed to be true?)

Reply:
The idea was that dereferencing the pointer should cause the memory management unit to cause an exception, which isn't true when it's a valid pointer. How about something like `(char*)(((uintptr_t)nullptr) - 1)` (except with C++ style casts)?

Reply:
On the other hand, maybe simplicity wins the case here and we shouldn't aim for the exception?

Reply:
@tniessen Sure, anything works for me, and the suggestion is optional anyway :)